src/ds/ptrwrap.h

```diff
@@ -101,14 +101,20 @@ namespace snmalloc
   template<typename T, capptr_bounds bounds>
   struct CapPtr
   {
-    T* unsafe_capptr;
+    uintptr_t unsafe_capptr;
```

Comment:
I'm a little bit nervous about this from a non-CHERI codegen perspective.  Casting a pointer to and from an integer type adds it to the may-alias-anything set.  Did you check whether this affected codegen for x86?